### DISCUSSION TITLE
Update persistent state when initializing

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [pmndrs/zustand#1673] [Q&A]
I am using Zustand in React app with React Provider. I need to merge two states when initializing the store. But it doesn't merge. It keeps the prev state untouched. What am I missing here? here is an example of what I did.

```js
const createAStore = (props) => {
  const cached = localStorge.get("storage-key").key;
  const update = () => {
    // ...implemetation
  };
  
  const merged = update(props.list, cached);
  
  return createStore(
    persist(
      (set) => {
        return {
          list: merged, // it doesn't merged as expected,
          // other states 
        };
      },
      { name: "storage-key" }
    )
  );
};
```

**Comment by dai-shi** [comment]
Not sure, but you create a new store in render? You should memoize it to give a stable store reference. It's tricky, but `use-constant` is a safe bet. (maybe with ref={}, or useRef+useEffect if that doesn't work.)
